### PR TITLE
Add module_util support for transit gateway vpc attachment module

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY

This PR adds support waiter and functinality support for transit gateway vpc attachment module.
This is needed for https://github.com/ansible-collections/community.aws/pull/2157

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by alinabuzachis — APPROVED** [review]

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

File: plugins/module_utils/ec2.py [hunk 1]
```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
...
+    _CUSTOM_EXCEPTION = AnsibleEC2Error
+
+    @classmethod
+    def _is_missing(cls):
+        return is_boto3_error_code("InvalidGatewayID.NotFound")
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("describe transit gateway attachments")
+@AWSRetry.jittered_backoff()
+def describe_vpc_attachments(
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1357]:
```suggestion
def describe_transit_gateway_vpc_attachments(
```

File: plugins/module_utils/ec2.py [hunk 2]
```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
...
+    client, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
+) -> List:
+    paginator = client.get_paginator("describe_transit_gateway_vpc_attachments")
+    return paginator.paginate(**params).build_full_result()["TransitGatewayVpcAttachments"]
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("create transit gateway vpc attachment")
+@AWSRetry.jittered_backoff()
+def create_vpc_attachment(
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1366]:
```suggestion
def create_transit_gateway_vpc_attachment(
```

File: plugins/module_utils/ec2.py [hunk 3]
```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
...
+    client, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1367]:
```suggestion
    client, transit_gateway_id: str, vpc_id: str, subnet_ids: List[str], **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
```

File: plugins/module_utils/ec2.py [hunk 4]
```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
...
+) -> Optional[Dict[str, Any]]:
+    result = client.create_transit_gateway_vpc_attachment(**params)
+    return result.get("TransitGatewayVpcAttachment", None)
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1370]:
```suggestion
    return client.create_transit_gateway_vpc_attachment(**params)["TransitGatewayVpcAttachment"]
```

File: plugins/module_utils/ec2.py [hunk 5]
```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
...
+) -> List:
+    paginator = client.get_paginator("describe_transit_gateway_vpc_attachments")
+    return paginator.paginate(**params).build_full_result()["TransitGatewayVpcAttachments"]
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("create transit gateway vpc attachment")
+@AWSRetry.jittered_backoff()
+def create_vpc_attachment(
+    client, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
+) -> Optional[Dict[str, Any]]:
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1368]:
```suggestion
) -> Dict[str, Any]:
```

File: plugins/module_utils/ec2.py [hunk 6]
```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
...
+    result = client.create_transit_gateway_vpc_attachment(**params)
+    return result.get("TransitGatewayVpcAttachment", None)
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("modify transit gateway vpc attachment")
+@AWSRetry.jittered_backoff()
+def modify_vpc_attachment(
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1375]:
```suggestion
def modify_transit_gateway_vpc_attachment(
```

File: plugins/module_utils/ec2.py [hunk 7]
```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
...
+    client, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
+) -> Optional[Dict[str, Any]]:
+    result = client.modify_transit_gateway_vpc_attachment(**params)
+    return result.get("TransitGatewayVpcAttachment", None)
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1379]:
```suggestion
    client, transit_gateway_attachment_id: str, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
) -> Dict[str, Any]:
    return client.modify_transit_gateway_vpc_attachment(**params)["TransitGatewayVpcAttachment"]
```

File: plugins/module_utils/ec2.py [hunk 8]
```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
...
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.deletion_error_handler("delete transit gateway vpc attachment")
+@AWSRetry.jittered_backoff()
+def delete_vpc_attachment(
+    client, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
+) -> Optional[Dict[str, Any]]:
+    result = client.delete_transit_gateway_vpc_attachment(**params)
+    return result.get("TransitGatewayVpcAttachment", None)
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1388]:
```suggestion
def delete_transit_gateway_vpc_attachment(transit_gateway_attachment_id: str) -> bool:
client.delete_transit_gateway_vpc_attachment(TransitGatewayAttachmentId=transit_gateway_attachment_id)
    return True
```

**Review by abikouo — CHANGES_REQUESTED** [review]

File: plugins/module_utils/ec2.py (hunk 7 above)
Comment (alinabuzachis) [reply plugins/module_utils/ec2.py]: `transit_gateway_attachment_id` is already passed inside params.

File: plugins/module_utils/ec2.py [hunk 9]
```diff
@@ -1373,6 +1373,49 @@ def get_sg_id(sg):
...
+# EC2 Transit Gateway VPC Attachment Error handler
+class EC2TransitGatewayVPCAttachmentErrorHandler(AWSErrorHandler):
+    _CUSTOM_EXCEPTION = AnsibleEC2Error
+
+    @classmethod
+    def _is_missing(cls):
+        return is_boto3_error_code("InvalidGatewayID.NotFound")
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("describe transit gateway attachments")
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1385]:
```suggestion
@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("describe transit gateway vpc attachments")
```

**Review by abikouo — CHANGES_REQUESTED** [review]

**Review by abikouo — APPROVED** [review]

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
regate

**Comment by alinabuzachis** [comment]
regate

**Pull request merged**